contracts/nois-gateway/src/contract.rs

```diff
@@ -505,6 +506,44 @@ fn execute_add_verified_round(
         .add_attributes(attributes))
 }
 
+fn execute_cleanup(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    limit: Option<u32>,
+) -> Result<Response, ContractError> {
+    // the number of elements we can still take (decreasing over time)
+    let mut limit = limit.unwrap_or(20) as usize;
+
+    let mut deleted = 0;
+    const PER_SCAN: usize = 20;
+    loop {
+        let take_this_scan = std::cmp::min(PER_SCAN, limit);
+        let keys: Vec<_> = deps
+            .storage
+            .range_keys(
+                Some(DRAND_JOBS_V1_START),
+                Some(DRAND_JOBS_V1_END),
+                Order::Ascending,
+            )
+            .take(take_this_scan)
```

Comment:
Is there a way here to filter on jobs that have a round in the past or that have been processed. 
This would
- Add some safety so we don't delete unprocessed jobs.
- Maybe make this a permissionless function

Reply:
Yeah, it's possible by changing the end from `\x00\x18drand_jobs_up_9999999999` to something like `\x00\x18drand_jobs_up_0000012345`.

However, the testnet gateway does not write to this store since July 2nd and the mainnet gateway does not write to this store since July 29th. If we just deploy this for testnet now and wait 2 weeks until we deploy to main, we should be fine I suppose.

Reply:
Oh so this will only cleanup the old state because the new state is stored with a different key prefix ?

Reply:
Yeah, exactly

Reply:
So if it wasnt for has caluclations we could have made this function inside the migrate entrypoint, but because this needds to be executed mulitple times (and we donnnow the calulcation how many times it takes in gas) that we're gonna make an entrypoint out of it ?

Reply:
Since we used `Deque`s in drand_jobs1, we have `>= round_with_jobs * 3` storage entries in that range. This can easily be thousands of entries. Trying to do it in migrate would most likely cause an out of gas issue.

Reply:
Yeah that's what I thought. Makes total sense. Thanks for the explanation.